torch/_dynamo/variables/builtin.py

```diff
@@ -1075,7 +1075,11 @@ def call_type(self, tx, obj: VariableTracker):
                 self, obj
             )
 
-        unimplemented(f"type({obj})")
+        raise UserError(
+            UserErrorType.ANTI_PATTERN,
+            "Can't call type() on generated custom object. "
+            "Please use __class__ instead",
+        )
```

Comment:
Sorry I just have time to look at this, I think this is not correct. What I mean is ```type(objA).func``` is anti-pattern, but ```type(objA)``` is not anti-pattern, since users want to get the type of a user defined object, which is a reasonable use case. That's why I suggested to capture the whole byte code sequence and throw the ```UserError```.